### PR TITLE
[Validator] deprecate the loose e-mail validation mode

### LINKED ISSUE #41855: [Validator] Change default mode of `Assert\Email` to "strict", for consistency with Mailer
**Description**
Since Symfony Mailer is using `egulias/email-validator` to validate email addresses, I'm suggesting to make this the default for `Assert\Email`.

Why? When using a framework to let users enter their email address, my minimum expectation would be that the very same framework is able to send an email there ;-) In other words: Symfony needs to stay consistent.

See https://github.com/symfony/symfony/issues/41850 for an example I encountered in production - but I'm sure there are better ones ;-)

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | Fix #41855
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Validator/Tests/Constraints/EmailValidatorTest.php [hunk 1]
```diff
@@ -76,6 +79,23 @@ public function getValidEmails()
             ['fabien@symfony.com'],
             ['example@example.co.uk'],
             ['fabien_potencier@example.fr'],
+        ];
+    }
+
+    /**
+     * @group legacy
+     * @dataProvider getValidInLooseModeEmails
```
Comment (stof) [inline src/Symfony/Component/Validator/Tests/Constraints/EmailValidatorTest.php]: this should also have a second `@dataProvider getValidEmails`, to ensure that emails valid in HTML5 mode are still valid in loose mode.

Comment (xabbuh) [reply src/Symfony/Component/Validator/Tests/Constraints/EmailValidatorTest.php]: done, same for `testValidNormalizedEmailsInLooseMode()`

**Review by stof — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

File: src/Symfony/Component/Validator/Constraints/Email.php [hunk 2]
```diff
@@ -68,6 +71,10 @@ public function __construct(
         $this->mode = $mode ?? $this->mode;
         $this->normalizer = $normalizer ?? $this->normalizer;
 
+        if (self::VALIDATION_MODE_LOOSE === $this->mode) {
+            trigger_deprecation('symfony/validator', '6.2', 'The "%s" mode is deprecated. The default mode will be changed to "%s" in 7.0.', self::VALIDATION_MODE_LOOSE, self::VALIDATION_MODE_HTML5);
```
Comment (ghost) [inline src/Symfony/Component/Validator/Constraints/Email.php:75]: Will the default be changed or will the validation mode be removed?

File: src/Symfony/Component/Validator/Constraints/Email.php [hunk 3]
```diff
@@ -27,6 +27,9 @@ class Email extends Constraint
 {
     public const VALIDATION_MODE_HTML5 = 'html5';
     public const VALIDATION_MODE_STRICT = 'strict';
+    /**
+     * @deprecated since Symfony 6.2
```
Comment (ghost) [inline src/Symfony/Component/Validator/Constraints/Email.php:31]: @deprecated since Symfony 6.2,  use VALIDATION_MODE_HTML5 instead?

**Review by ghost — COMMENTED** [review]
@xabbuh i know the code is already merged but maybe you can have a look at my comments?

**Comment by xabbuh** [comment]
@johanadivare sorry for the late feedback, please see #49420

**Pull request merged**